### PR TITLE
hotshot update

### PR DESCRIPTION
**Pull request by tbro**
Update hotshot to tag 0.5.70. Relevant upstream changes are: https://github.com/EspressoSystems/HotShot/pull/3555.

### CODE REVIEW

File: types/src/v0/impls/header.rs [hunk 1]
```diff
@@ -425,7 +424,7 @@ impl Header {
             fee_amount,
         } in &builder_fee
         {
-            if version < MarketplaceVersion::VERSION {
+            if version < StaticVersion::<0, 3>::version() {
```
Comment (tbro) [inline types/src/v0/impls/header.rs:427]: @ss-es @imabdulbasit was `MarketplaceVersion` replaced or moved elsewhere?

Comment (imabdulbasit) [reply types/src/v0/impls/header.rs:427]: its a type now in Versions trait

File: types/src/v0/mod.rs [hunk 2]
```diff
@@ -130,14 +133,22 @@ impl NodeType for SeqTypes {
...
+impl Versions for SequencerVersions {
     type Base = StaticVersion<0, 1>;
     type Upgrade = StaticVersion<0, 2>;
-    type AuctionResult = SolverAuctionResults;
     const UPGRADE_HASH: [u8; 32] = [
         1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
         0, 0,
     ];
+
+    type Marketplace = StaticVersion<0, 3>;
```
Comment (imabdulbasit) [inline types/src/v0/mod.rs:149]: here

File: types/src/v0/impls/header.rs (hunk 1 above)
Comment (tbro) [reply types/src/v0/impls/header.rs]: 66f8d00e912e41a7702befa902d5e26f67da8681

**Review by imabdulbasit — APPROVED** [review]

**Pull request merged**